src/txn/txn_ckpt.c

```diff
@@ -1141,9 +1142,20 @@ __drop(WT_CKPT *ckptbase, const char *name, size_t len)
      * it's one we want to drop.
      */
     if (strncmp(WT_CHECKPOINT, name, len) == 0) {
+        /*
+         * Currently, hot backup cursors block checkpoint drop, which means releasing a hot backup
+         * cursor can result in immediately attempting to drop lots of checkpoints, which involves a
+         * fair amount of work while holding locks. Limit the number of standard checkpoints dropped
+         * per checkpoint.
+         */
+        max_ckpt_drop = 0;
         WT_CKPT_FOREACH (ckptbase, ckpt)
-            if (WT_PREFIX_MATCH(ckpt->name, WT_CHECKPOINT))
+            if (WT_PREFIX_MATCH(ckpt->name, WT_CHECKPOINT)) {
                 F_SET(ckpt, WT_CKPT_DELETE);
+#define WT_MAX_CHECKPOINT_DROP 4
```

Comment:
Why we make this change here?

Reply:
That is weird, it is last commit already in develop:
https://github.com/wiredtiger/wiredtiger/commit/a19d2bff3d30fdd83086172b29161527aefc28e3

Not sure why this is showing as part of this diff.

Reply:
Seems to have fixed by doing a merge of the develop into the change.